src/validation.cpp

```diff
@@ -1595,6 +1595,13 @@ static unsigned int GetBlockScriptFlags(const CBlockIndex* pindex, const Consens
     {
         // Enforce WITNESS rules whenever P2SH is in effect
         flags |= SCRIPT_VERIFY_P2SH | SCRIPT_VERIFY_WITNESS;
+        // Enforce Taproot (BIP340-BIP342) whenever WITNESS is in effect,
+        // unless except
+        if (!consensusparams.TaprootException.has_value() ||          // no Taproot exception on this chain
```

Comment:
Can just be `!consensusparams.TaprootException` (https://en.cppreference.com/w/cpp/utility/optional/operator_bool).

Reply:
Gone